linker/jvm/src/test/scala/org/scalajs/linker/GCCLinkerTest.scala

```diff
@@ -30,4 +34,27 @@ class GCCLinkerTest {
      */
     testLink(Nil, Nil, config = StandardConfig().withClosureCompiler(true))
   }
+
+  @Test
+  def linkIncrementalSmoke(): AsyncResult = await {
+    /* Check that linking twice works. GCC trees are highly mutable, so if we
+     * (re-)use them wrongly over multiple runs, things can fail unexpectedly.
+     */
+    val classDefs = MemClassDefIRFile(
+        mainTestClassDef(consoleLog(str("test")))) :: Nil
+
+    val moduleInitializers = MainTestModuleInitializers
+
+    val config = StandardConfig().withCheckIR(true).withClosureCompiler(true)
+    val linker = StandardImpl.linker(config)
+
+    val output = MemOutputDirectory()
+    val logger = new ScalaConsoleLogger(Level.Error)
+
+    for {
+      lib <- TestIRRepo.minilib
+      _ <- linker.link(lib ++ classDefs, moduleInitializers, output, logger)
+      _ <- linker.link(lib ++ classDefs, moduleInitializers, output, logger)
```

Comment:
We probably want to change *something* in between the two runs. Perhaps the text within the `consoleLog`. Otherwise we might have a global shortcut that avoids doing any work at all.

Reply:
Good point. Updated.

(I have also avoided the additional list for `classDefs`, seemed simpler).